docs/proposals/integrations_release_cycle.md

```diff
@@ -0,0 +1,63 @@
+# Integrations release cycle
+
+- Authors: Massimiliano Pippi
+- Date: 2018-12-12
+- Status: draft
+- [Discussion](https://github.com/DataDog/integrations-core/pull/2741)
+
+## Overview
+
+Each agent based integration has its own version number and can be released as
+a Python package at any given time. This document outlines what the release
+cycle will be for each integration.
+
+## Problem
+
+At this moment the release cycle for most of the integration is not formalized
+and there's no process in place to keep it consistent and reliable. Once a feature
+or a bugfix is merged on `master`, different things may happen:
+
+- A release is immediately triggered.
+- A release is triggered after merging other, maybe unrelated changes for the same integration.
+- A release is triggered at a later time without any specific deadline.
+
+This causes a number of issues for all actors involved:
+
+- Engineers have to improvise on a case by case basis.
+- Contributors have no idea if changes within a PR will be released and when.
+- Users don't know when a feature or a bugfix will be actually available.
+
+## Constraints
+
+1. Each integration must keep its own release cycle.
+2. The solutions must work for integrations from `integrations-extra` as well.
+
+## Recommended Solution
+
+A release must be immediately triggered after merging a PR into the `master` branch.
+To ensure quality of the software shipped, **changes on the branch should be tested before merging**.
+A pre-release testing checklist is out of the scope of this RFC but in general
```

Comment:
How much of the testing here is fully automated?

Reply:
- validate the list of checks shipped with the agent is correct
- validate config files are well formed (YAML, presence of an instance, etc)
- validate the dependencies declared in the check are compatible with what the Agent ships
- validate manifest file is valid (both JSON format and contents)
- validate metadata (same as manifest)
- validate service-checks (same as manifest)
- unit and integrations test + test coverage (no agent involved)
- benchmarks (at the moment they don't fail but we could stop a release if a slowdown is found)

The list can be found here: https://github.com/DataDog/integrations-core/blob/master/.travis.yml#L50-L57